test-case/check-s3.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+##
+## Case Name: check S3 status
+## Preconditions:
+##    N/A
+## Description:
+##    Run the S3 command to check device status
+## Case step:
+##    1. switch S3 operation
+##    2. use rtcwake -m mem command to do S3
+##    3. check command return value
+##    4. check dmesg errors
+##    5. check wakeup increase
+## Expect result:
+##    S3 recover
+##    check kernel log and find no errors
+##
+
+source $(dirname ${BASH_SOURCE[0]})/../case-lib/lib.sh
+
+# wait time should >= 3 for other device wakeup from sleep
+random_min=3
+random_max=20
+
+OPT_OPT_lst['l']='loop'     OPT_DESC_lst['l']='loop count'
+OPT_PARM_lst['l']=1         OPT_VALUE_lst['l']=5
+
+OPT_OPT_lst['p']='power'    OPT_DESC_lst['p']="S3 power type from /sys/power/mem_sleep"
+OPT_PARM_lst['p']=1         OPT_VALUE_lst['p']="s2idle"
+
+OPT_OPT_lst['s']='sleep'    OPT_DESC_lst['s']='S3 command:rtcwake sleep option'
+OPT_PARM_lst['s']=1         OPT_VALUE_lst['s']=5
+
+OPT_OPT_lst['w']='wait'     OPT_DESC_lst['w']='wait time after S3 wakeup'
+OPT_PARM_lst['w']=1         OPT_VALUE_lst['w']=5
+
+OPT_OPT_lst['r']='random'   OPT_DESC_lst['r']="Random setup wait/sleep time, time range is [$random_min-$random_max], this option will overwait s & w option"
+OPT_PARM_lst['r']=0         OPT_VALUE_lst['r']=0
+
+func_opt_parse_option $*
+func_lib_check_sudo
+func_lib_setup_kernel_last_line
+
+power=${OPT_VALUE_lst['p']}
+dlogi "Current power mode: $(cat /sys/power/mem_sleep)"
+# check for power value effect
+[[ ! "$(cat /sys/power/mem_sleep|grep $power)" ]] && dloge "useless power option" && exit 2
+dlogc "sudo bash -c 'echo $power > /sys/power/mem_sleep'"
+sudo bash -c "'echo $power > /sys/power/mem_sleep'"
+
+loop_count=${OPT_VALUE_lst['l']}
+declare -a sleep_lst wait_lst
+
+if [ ${OPT_VALUE_lst['r']} -eq 1 ]; then
```

Comment:
can we make the random number generator into a help function.

Reply:
OK, I will add this help function